src/ansys/fluent/core/services/datamodel_se.py

```diff
@@ -1299,17 +1309,25 @@ def __init__(
         else:
             self.path = path
 
-    def __call__(self, *args, **kwds) -> Any:
+    def __call__(self, purpose: Optional[str] = None, *args, **kwds) -> Any:
         """Execute the command.
 
         Returns
         -------
         Any
             Return value.
         """
-        return self.service.execute_command(
+        if "Read" in self.__class__.__name__ and bool(
```

Comment:
@hpohekar We need to eliminate these checks for "Read" and "Write" in the runtime code. We should be generating different types in the codegen instead.

Reply:
Yes, as mentioned here, it's temporary. We are working on meshing datamodel static info. We will update this soon.

Reply:
We have removed these checks. Now we are extracting file_purpose from datamodel static info.